src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSerializationVisitor.cs

```diff
@@ -145,7 +145,7 @@ protected override void ProcessEntityType(IEdmEntityType element)
             this.BeginElement(element, this.schemaWriter.WriteEntityTypeElementHeader);
             if (element.DeclaredKey != null && element.DeclaredKey.Any())
             {
-                this.VisitEntityTypeDeclaredKey(element.DeclaredKey);
+                this.VisitEntityTypeDeclaredKey(element.DeclaredKey, element.DeclaredKeyAlias);

```

Comment:
Does this mean that you have separate properties for the key properties that are defined on the entity (structural properties) and the key properties that are aliases to properties on complex or related types?

Reply:
Key properties are structural properties within an entity type (in which they are declared) that have been declared as key properties. The DeclaredKeyAliases has a mapping between these KeyProperties and their aliases. I don't think we can have a situation where we have aliases but no key properties but the reverse is possible.